llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1250,6 +1268,158 @@ void AddressSanitizerPass::printPipeline(
   OS << '>';
 }
 
+// Append a new argument "launch_data" to user's spir_kernel & spir_func
+static void ExtendSpirKernelArgs(Module &M, FunctionAnalysisManager &FAM) {
+  SmallVector<Function *> SpirFixupFuncs;
+  for (Function &F : M) {
+    if (F.getCallingConv() == CallingConv::SPIR_KERNEL) {
+      SpirFixupFuncs.emplace_back(&F);
+    } else if (F.getCallingConv() == CallingConv::SPIR_FUNC) {
+      if (M.getModuleFlag("openmp-device")) {
+        const TargetLibraryInfo &TLI = FAM.getResult<TargetLibraryAnalysis>(F);
+        LibFunc LF;
+        if (TLI.getLibFunc(F, LF)) {
+          continue;
+        } else {
+          if (F.getName().starts_with("_Z") &&
+              F.getName().find("__spirv_") != StringRef::npos)
+            continue;
+
+          if (F.getName().equals("_Z12get_local_idj") ||
+              F.getName().equals("_Z14get_local_sizej") ||
+              F.getName().equals("_Z14_get_num_groupsj"))
+            continue;
+        }
+
+      } else {
+        if (F.isDeclaration() || F.getName().starts_with("__sycl_") ||
+            F.getName().starts_with("__spirv") ||
+            F.getFunctionType()->isVarArg())
+          continue;
+      }
+      SpirFixupFuncs.emplace_back(&F);
```

Comment:
we can probably also skip function that [doesNotAccessMemory](https://llvm.org/doxygen/classllvm_1_1Function.html#aa701465fb174e27e3a89695762970997)

Reply:
done. move this into SanitizeExtendArgumentPass.